### PR TITLE
Add _middleware support

### LINKED ISSUE #64: Add `_middleware.ts` file
### What is the feature you are proposing?

Would it be possible to create a `_middleware.ts` file that would automatically add the middleware to routes at and below it? Cumulative `_middleware` declarations would be merged, with higher ones taking precedence in the ordering.

```ts
//_middleware.ts
import { someMiddleware } from '../middleware/someMiddleware'
import { someOtherMiddleware } from '../middleware/someOtherMiddleware'

// middleware would be ran in the order declared here
exports default [ someMiddleware, someOtherMiddleware ]
```

This would make it unnecessary to have to redeclare the middleware pipeline for each route, or manually create a Hono router just to gain access to the `use` method.

Directory structure would look like:
```sh
.
└── app/
    └── routes/
        ├── some-path/
        │   ├── uses-middleware/
        │   │   └── index.ts    <--uses _middleware
        │   ├── _middleware.ts
        │   └── index.ts        <--uses _middleware
        └── index.ts
```

### PR DESCRIPTION
**Pull request by bruceharrison1984**
The `_middleware` file allows you to specify middleware at any level in the directory hierarchy. 

```ts
//_middleware.ts
import { someMiddleware } from '../middleware/someMiddleware'
import { someOtherMiddleware } from '../middleware/someOtherMiddleware'

// middleware would be ran in the order declared here
exports default [ someMiddleware, someOtherMiddleware ]
```

It is essentially a shortcut for the `hono.use` method, and it will apply the middleware to all descendant routes. This effect is still cumulative, so parent middleware is also applied.

Integration tests were added, and a bit of cleanup/consolidation with the `_renderer` logic was done because both things are initialized in essentially the same way.

This would close #64

### CODE REVIEW

**Comment by yusukebe** [comment]
Thanks! I'll comment later👍

**Comment by yusukebe** [comment]
Hi @bruceharrison1984 !

Awesome! I think this will make HonoX much better!

I have two things we want to consider.

### 1. We can allow `middleware.ts` not only `middleware.tsx`.

Currently, your implementation only allows `.tsx` extensions, but what about allowing `.ts` as well? Both are allowed, but since we don't use JSX much in middleware, so I think we can recommend using `.ts`.

### 2. How about recommend using `createRoute()`?

The following writing style is indeed not bad.

```ts
export default [middleware1, middleware2]
```

However, I prefer to use `createRoute()` imported from `honox/factory`.

```ts
import { createRoute } from 'honox/factory'
//...
export default createRoute(middlware1, middleweare2)
```

This gives it type support. For example, it is very useful when using Validator.

Screenshot 2024-02-19 at 14 51 17

Of course, either way is acceptable, but I think the better way to put it in the README is to use `createRoute()`.

---

How do you think about these?

**Comment by bruceharrison1984** [comment]
@yusukebe I agree on both counts, especially the second one because it gives some much needed structure to an otherwise mysterious magic export. I can get these changes implemented this week, perhaps later today.

**Comment by yusukebe** [comment]
@bruceharrison1984 

Thanks for your super quick response! Okay.

**Comment by bruceharrison1984** [comment]
@yusukebe I couldn't sleep, so changes are implemented as discussed 😄

File: README.md [hunk 1]
```diff
@@ -466,6 +466,25 @@ export const POST = createRoute(zValidator('form', schema), async (c) => {
 })
 ```
 
+Alternatively, you can use a `_middleware.(ts|tsx)` file in a directory to have that middleware applied to the current route, as well as all child routes. Middleware is ran in the order that it is listed within the array.
+
+An equivilant example to the previous Hono-style middleware configuration is as follows:
+
+```tsx
+// /app/_middleware.tsx
```
Comment (yusukebe) [inline README.md:474]: I know it's a small detail, but wouldn't `/app/_middleware.ts` be better here?

**Review by yusukebe — COMMENTED** [review]
@bruceharrison1984 

I've left one comment! 
Good night!

File: README.md (hunk 1 above)
Comment (bruceharrison1984) [reply README.md]: fixed!

**Comment by yusukebe** [comment]
@bruceharrison1984 

Opps. Could you run the fomatter and linter?

**Pull request merged**